packages/protocol/contracts/L1/StakingProverPool3.sol

```diff
@@ -0,0 +1,316 @@
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import {AddressResolver} from "../common/AddressResolver.sol";
+import {EssentialContract} from "../common/EssentialContract.sol";
+import {TaikoToken} from "./TaikoToken.sol";
+import {Proxied} from "../common/Proxied.sol";
+
+contract ProverPoolImpl is EssentialContract {
+    struct Staker {
+        uint8 id; // [0-31]
+        address prover;
+        uint32 stakedAmount; // unit is 10^8, this means a max of
+        // 429496729500000000 tokens, 2.3283%  of total supply
+        uint16 rewardPerGas; // expected rewardPerGas
+        uint16 capacity; // maximum capacity (parallel blocks)
+    }
+
+    // Then we define a mapping from id => Staker
+    mapping(uint256 id => Staker) stakers;
+    // We need an address to id mapping
```

Comment:
since we overwrite entries here, how does one get their staked info? Dont we need a mapping that just has ALL stakers in it, ie: `mapping(address staker => Staker stakerInfo) stakerInfos` so a user can pull down their current staked info?